app/scripts/controller-init/confirmations/transaction-controller-init.ts

```diff
@@ -128,6 +131,19 @@ export const TransactionControllerInit: ControllerInitFunction<
     // @ts-expect-error Controller uses string for names rather than enum
     trace,
     hooks: {
+      ///: BEGIN:ONLY_INCLUDE_IF(institutional-snap)
+      beforePublish: (transactionMeta) => {
```

Comment:
Going forward, I think best practice is to avoid any new hard references to controllers so could we achieve this via a messenger call such as `DeferredPublicationController:beforePublishHook`?

Reply:
Added here

https://github.com/MetaMask/metamask-extension/pull/30180/files#diff-601927d55129de0a1f694108115de7e4bfb29aa2626858116d3042099d82567eR11